connectors/sources/network_drive.py

```diff
@@ -706,7 +735,10 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: Dictionary containing the Network Drive files and folders as documents
         """
-
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(
+            executor=None, func=self.smb_connection.create_connection
+        )
```

Comment:
So if ping is called before, then it will create a different connection?

Reply:
Added this to make ping and get_docs independent of each other. `create_connection` works as `get_connection` method you mentioned in earlier comment